WordPress/Classes/Services/PostService+JetpackSocial.swift

```diff
@@ -0,0 +1,109 @@
+extension PostService {
+    typealias StringDictionary = [String: String]
+    typealias Keys = Post.Constants
+    typealias SkipPrefix = Post.PublicizeMetadataSkipPrefix
+
+    /// Returns a dictionary format for the `Post`'s `disabledPublicizeConnection` property based on the given metadata.
+    ///
+    /// This will try to handle both Publicize skip key formats, `_wpas_skip_{keyringID}` and `_wpas_skip_publicize_{connectionID`.
+    ///
+    /// There's a possibility that the `keyringID` obtained from remote doesn't match with any of the `PublicizeConnection`
+    /// that's stored locally, perhaps due to the app being out of sync. In this case, we'll fall back to using the old format.
+    ///
+    /// - Parameters:
+    ///   - post: The associated `Post` object. Optional because Obj-C shouldn't be trusted.
+    ///   - metadata: The metadata dictionary for the post. Optional because Obj-C shouldn't be trusted.
+    /// - Returns: A dictionary for the `Post`'s `disabledPublicizeConnections` property.
+    @objc(disabledPublicizeConnectionsForPost:andMetadata:)
+    func disabledPublicizeConnections(for post: AbstractPost?, metadata: [[String: Any]]?) -> [NSNumber: StringDictionary] {
+        guard let post,
+              let metadata = metadata as? [[String: String]] else {
+            return [:]
+        }
+
+        return metadata
+            .filter { $0[Keys.publicizeKeyKey]?.hasPrefix(SkipPrefix.keyring.rawValue) ?? false }
```

Comment:
Would this possibly filter out metadata which have both an array and the publicize skip info in it?

What about something like this?
```suggestion
        guard let post, let metadata else {
            return [:]
        }

        return metadata
            .compactMap { $0 as? [String: String] }
            .filter { $0[Keys.publicizeKeyKey]?.hasPrefix(SkipPrefix.keyring.rawValue) ?? false }
```

Reply:
Good point! Thanks Chris.